[core/dr] remove run shared folders

kidanger opened the pull request:
Input files are passed during the /exec_and_wait HTTP request to the DR, and the DR now returns a zip containing everything, including a `exec_info.json` file that corresponds to the old DR payload.

mcolom reviewed (approved):
About the review of the code, not much to say. Just add in demorunner.py a comment explaining that you're creating that exec_info.json which contains (...), and another comment in the block of code which follows, explaining that you're creating a ZIP file with the contents of the working folder. This way it's easier to follow the code, just reading the comments. Even if the code might be self-explained, it's faster if you can just read the comments.

I see that you also added a "# TODO: what is this?". If it's something you don't know, it's better that you see with Héctor and Jyotsna instead. If it's something wrong, we'll fix it or add an explanation comment in the code, if needed. You can remove the message meanwhile.

If you already tested this and you didn't observe any performance of other problems, you can merge this code in the devel branch and start testing in Integration.

kidanger commented:
Thanks, I added the two comments and removed the TODO.

Changes were tested on a private deployment but since it does not rely on sshfs, performances cannot really be measured against it. For local execution (core and a DR on the same host), the new changes add an overhead. We can measure this overhead in integration, but again the comparison with sshfs can only be done on core if we want to.
